config/webpack/production.js

```diff
@@ -44,8 +45,27 @@ module.exports = merge(sharedConfig, {
   plugins: [
     new CompressionPlugin({
       filename: '[path][base].gz[query]',
+      algorithm: 'gzip',
       cache: true,
       test: /\.(js|css|html|json|ico|svg|eot|otf|ttf|map)$/,
+      threshold: 256,
+      minRatio: 1,
+      compressionOptions: {
+        level: 9,
+      },
```

Comment:
Why those changes?

Reply:
To force necessary files to be packed into a gz archive. Subsequently, with the active option `gzip_static on`; nginx will immediately give compressed file.

Reply:
I know what the `CompressionPlugin` is for, I'm asking you why you changed the values. From my understanding, they differ from the default values in the following way:
- `threshold` has been changed from `0` to `256`
- `minRatio` has been changed from `0.8` (20% size reduction) to `1`

I see no harm in those (though `minRatio` of 1 means extra work for no benefits), but I don't see why they'd need to be changed either.

The other values (`algorithm: 'gzip' and `compressionOptions: { level: 9 }`) are the default ones.

Reply:
Removed defaults values.

> I see no harm in those (though minRatio of 1 means extra work for no benefits), but I don't see why they'd need to be changed either.

The compression process occurs only during build of application; no extra resources are wasted during working app. 

Updated PR.

Reply:
If file is not precompressed, then it will be dynamically compressed by server each time it is requested by client.

Reply:
Example.
With minRatio 0.8:
```
curl -k --head -H "Accept-Encoding: gzip" https://mastodon.local/packs/js/mailer-a69e593e827d2e39f347.chunk.js

HTTP/2 200
server: nginx
date: Tue, 23 Aug 2022 16:36:24 GMT
content-type: application/javascript
vary: Accept-Encoding
etag: W/"z6rcr06c0684i1wxisrqxayswx0ma9j0"
content-encoding: gzip
```
```
curl -k --head -H "Accept-Encoding: gzip" https://mastodon.local/packs/js/mailer-a69e593e827d2e39f347.chunk.js --output chunk.js.gz
ls -l chunk.js.gz
-rw-r--r-- 1 user users 196 Aug 23 16:36 chunk.js.gz
```
With minRatio 1:
```
curl -k --head -H "Accept-Encoding: gzip" https://mastodon.local/packs/js/mailer-a69e593e827d2e39f347.chunk.js

HTTP/2 200
server: nginx
date: Tue, 23 Aug 2022 16:27:49 GMT
content-type: application/javascript
content-length: 179
vary: Accept-Encoding
etag: "lx57dh0hi23n26d51big5iws7hx9jviw"
content-encoding: gzip
```
The difference, of course, is insignificant.

Reply:
Change to `minRatio 0.8`?

Reply:
Removing `minRatio 1`.